src/shared/conf-parser.c

```diff
@@ -1185,3 +1185,36 @@ int config_parse_permille(const char* unit,
 
         return 0;
 }
+
+int config_parse_vlanprotocol(const char* unit,
+                              const char *filename,
+                              unsigned line,
+                              const char *section,
+                              unsigned section_line,
+                              const char *lvalue,
+                              int ltype,
+                              const char *rvalue,
+                              void *data,
+                              void *userdata) {
+
+        int* vlanprotocol = data;
+        int proto = ETH_P_8021Q;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+
+        if (streq(lvalue, "802.1ad") || streq(lvalue, "802.1AD")) {
+          proto = ETH_P_8021AD;
+        } else if (streq(lvalue, "802.1q") || streq(lvalue, "802.1Q")) {
+          proto = ETH_P_8021Q;
+        } else {
+          assert_not_reached(lvalue);
```

Comment:
`assert_not_reached` is for places in code that should never be reached. It must not be used for verification of parameters specified by the user.